core/configuration.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+
+import os
+import sys
+import json
+import datetime
+from calendar import monthrange
+
+
+class Configuration(object):
+    """The Configuration class validates and prepares configuration data from
+    the config file for further use in the script.
+
+    """
+
+    def __init__(self, args):
+        """Contructor function
+
+        This function gets called when Configuration object are created.
+
+        Based on the configuration from the config file it validates and
+        prepares all mandatory configuration elements.
+
+        """
+        # Load config file from argument of standard location
+        self.config = self._load_config(args)
+
+        # Define name for output file
+        self.output = self._define_output_file(args)
+
+        # Validate and prepare start and end date
+        self._prepare_dates()
+
+    def _load_config(self, args):
+        """Loads the configuration. Either the standard location
+        (config.json) or a location specified as an command argument.
+
+        :return config: A dictionary of configuration data.
+
+        """
+        # Load config json file
+        if args.config is not None:
+            config = self._load_config_file(args.config)
+        elif os.path.isfile('config.json'):
+            with open("config.json") as json_file:
+                config = self._load_config_file(json_file)
+        else:
+            sys.stderr.write("Error: No config.json file found.\n")
+            sys.exit(0)
+
+        return config
+
+    def _load_config_file(self, configfile):
+        """
+        Loads json from config file
+
+        :return config: A dictionary of configuration data.
+
+        """
+        try:
+            config = json.load(configfile)
+        except ValueError, e:
+            sys.stderr.write('Error: Config json file is invalid.\n')
+            print(e)
+            sys.exit(0)
+
+        return config
+
+    def _define_output_file(self, args):
+        """
+        Defines the filename for GTFS file to write.
+        Either from config file or command argument.
+
+        :return output: Filename for saving GTFS output file.
+
+        """
+        # Get and check filename for gtfs output
+        if args.output is not None:
+            output_file = args.output
+        elif 'output_file' in self.config:
+            output_file = self.config['output_file']
+        else:
+            sys.stderr.write('Error: No filename for gtfs file specified.\n')
+            sys.exit(0)
+
+        return output_file
+
+    def _prepare_dates(self):
+        """
+        Validate and prepare start and end date.
+        Either from config file or based on current date.
+
+        """
+        config = self.config
+
+        try:
+            start_date = datetime.datetime.strptime(
+                config['feed_info']['start_date'], "%Y%m%d")
+        except:
```

Comment:
Exceptions were working very well for me. What's the issue? I don't like the if clauses because we have to check for two things `if 'start_date' in config['feed_info']` and if `datetime.datetime.strptime(config['feed_info']['start_date'], "%Y%m%d")` works successfully.

I'm also very interested if you could describe please what you mean with better style! I like to know how to improve code style! Thank you!